classes/local/store/object_file_system.php

```diff
@@ -316,6 +322,47 @@ public function delete_object_from_local_by_hash($contenthash, $objectsize = 0)
         return $finallocation;
     }
 
+    /**
+     * Recursively reads dirs from passed path and delete all empty dirs.
+     * @param string $rootpath Full path to the dir.
+     */
+    public function delete_empty_folders($rootpath) {
+        if (!is_dir($rootpath)) {
+            return;
+        }
+        $iterator = new RecursiveDirectoryIterator($rootpath);
+        $iterator->setFlags(RecursiveDirectoryIterator::SKIP_DOTS);
+        $directories = new ParentIterator($iterator);
+        foreach (new RecursiveIteratorIterator($directories, RecursiveIteratorIterator::CHILD_FIRST) as $dir) {
+            $dirpath = $dir->getPathname();
+            if ($this->is_dir_empty($dirpath)) {
+                rmdir($dirpath);
+            }
+        }
+        // Make sure we keep the 'filedir' dir.
+        if ('filedir' !== basename($rootpath) && $this->is_dir_empty($rootpath)) {
```

Comment:
Can you test filedir without this hard coded, eg $CFG->filedir which should be setup earlier?

Reply:
updated